torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1921,33 +1922,59 @@ def _full_post_state_dict_hook(
         back to sharded version after _summon_full_params ends, and also remove
         "_fsdp_wrapped_module" prefix.
         """
+        _replace_by_prefix(state_dict, prefix + f"{FSDP_WRAPPED_MODULE}.", prefix)
+
         self._assert_state([TrainingState_.SUMMON_FULL_PARAMS])
         # state_dict is empty for nonzero ranks if `rank0_only` was enabled.
         if not state_dict:
             return state_dict
 
+        if self._fsdp_wrapped_module.no_params:
+            return state_dict
+
+        # This can happen when rank0_only is enabled and self.rank != 0.
+        if self._fsdp_wrapped_module.orig_flat_param[0] is None:
+            assert self._fsdp_wrapped_module.flat_param is not None, (
+                "When no_params is False, one of flat_param and orig_flat_param "
```

Comment:
Is this a developer or user facing issue? If user can hit this issue, might be good to mention explicitly what went wrong and raise a Runtime/ValueError.

Reply:
It should be internal.

Reply:
Should be an assert here as this should not happen (anymore). If it happens, it should be an assert.